crates/isograph_compiler/src/watch.rs

```diff
@@ -52,20 +64,106 @@ pub async fn handle_watch_command(
     .await
 }
 
-fn any_modified_path_is_outside_artifact_directory(
+fn changed_files(
     events: &[DebouncedEvent],
-    artifact_directory: &PathBuf,
-) -> bool {
-    for event in events.iter() {
-        for watched_path in event.paths.iter() {
-            // These paths are canonicalized, so it is sufficient
-            // to check starts_with
-            if !watched_path.starts_with(artifact_directory) {
-                return true;
+    config: &CompilerConfig,
+) -> Option<Vec<SourceFileEvent>> {
+    let mut source_file_events = vec![];
+    for event in events {
+        if let Some(source_file_event) = match event.kind {
+            EventKind::Create(create_kind) => {
+                process_create_event(config, create_kind, &event.paths)
+            }
+            EventKind::Modify(modify_kind) => {
+                process_modify_event(config, modify_kind, &event.paths)
             }
+            EventKind::Remove(remove_kind) => {
+                process_remove_event(config, remove_kind, &event.paths)
+            }
+            _ => None,
+        } {
+            source_file_events.push(source_file_event);
         }
     }
-    false
+    if source_file_events.is_empty() {
+        None
+    } else {
+        Some(source_file_events)
+    }
+}
+
+fn process_create_event(
+    config: &CompilerConfig,
+    create_kind: CreateKind,
+    paths: &[PathBuf],
+) -> Option<SourceFileEvent> {
+    match create_kind {
+        CreateKind::File if paths.len() == 1 => changed_file_kind(config, &paths[0])
```

Comment:
can you add a `TODO handle symlink creation`? Likewise, for symlink deletion and renaming? (Those are `CreateKind::Other` or `RemoveKind::Other`). 

Anyway, the compiler does not ignore symlinks, so we should handle them. But we can just note it for now, I don't want to worry about this, since it's obviously silly